test/util/IdTableHelpers.cpp

```diff
@@ -181,27 +162,48 @@ IdTable createRandomlyFilledIdTable(const size_t numberRows,
 }
 
 // ____________________________________________________________________________
-IdTable createRandomlyFilledIdTable(const size_t numberRows,
-                                    const size_t numberColumns,
-                                    const size_t joinColumn,
-                                    const size_t joinColumnLowerBound,
-                                    const size_t joinColumnUpperBound) {
+IdTable createRandomlyFilledIdTable(
+    const size_t numberRows, const size_t numberColumns,
+    const std::vector<JoinColumnAndBounds>& joinColumnsAndBounds) {
   // Entries in IdTables have a max size.
   constexpr size_t maxIdSize = ValueId::maxIndex;
 
-  // Is the lower bound smaller, or equal, to the upper bound?
-  AD_CONTRACT_CHECK(joinColumnLowerBound <= joinColumnUpperBound);
-  // Is the upper bound smaller, or equal, to the maximum size of an IdTable
-  // entry?
-  AD_CONTRACT_CHECK(joinColumnUpperBound <= maxIdSize);
-
-  // The random number generators for join column entries.
-  SlowRandomIntGenerator<size_t> joinColumnEntryGenerator(joinColumnLowerBound,
-                                                          joinColumnUpperBound);
+  /*
+  Is the lower bound smaller, or equal, to the upper bound? And is the upper
+  bound smaller, or equal, to the maximum size of an IdTable entry?
+  */
+  AD_CONTRACT_CHECK(std::ranges::all_of(
+      joinColumnsAndBounds, [](const JoinColumnAndBounds& j) {
+        return j.lowerBound <= j.upperBound && j.upperBound <= maxIdSize;
+      }));
 
-  return createRandomlyFilledIdTable(
-      numberRows, numberColumns, {joinColumn}, [&joinColumnEntryGenerator]() {
-        // `IdTable`s don't take raw numbers, you have to transform them first.
-        return ad_utility::testing::VocabId(joinColumnEntryGenerator());
+  /*
+  Instead of doing things ourselves, we will call the overload, who takes pairs
+  of column indexes and functions.
+  */
+  const auto& joinColumnAndGenerator = ad_utility::transform(
+      joinColumnsAndBounds, [](const JoinColumnAndBounds& j) {
+        return std::make_pair(
+            j.joinColumn,
+            // We don't use a static variable for generator, because static
+            // variables are shared between different instances of the lambda.
```

Comment:
You can just write `// Each column gets its own random generator`. (Shorter and sufficient.)

Reply:
Done, see [e290909](https://github.com/ad-freiburg/qlever/pull/1016/commits/e290909efd3c584c1e8b42f67f6740729961503d).